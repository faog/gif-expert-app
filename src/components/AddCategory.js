import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (evt) => {
    setInputValue(evt.target.value);
  };

  const handleSubmit = (evt) => {
    evt.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories(categories => [inputValue, ...categories]);
      setInputValue('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <p>{ inputValue }</p>
      <input
        type="text"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
};
