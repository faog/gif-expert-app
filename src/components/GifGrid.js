import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async() => {
    const url = 'https://api.giphy.com/v1/gifs/search?api_key=SehYrvVMKCk9VRuh9nyEK2w66JLEITpL&q=onepunchman&limit=10';
    const resp = await fetch(url);
    const { data } = await resp.json();

    const gifs = data.map(img => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url
      };
    });

    setImages(gifs);
  };

  return (
    <div>
      <h3>{ category }</h3>
      {
        images.map(img =>
          <GifGridItem
            key={img.id}
            {...img}
          />
        )
      }
    </div>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string
};
