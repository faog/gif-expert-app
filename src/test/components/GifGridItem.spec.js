import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('<GifGridItem /> Component', () => {
  let wrapper;

  const props = {
    title: 'Un titulo',
    url: 'https://localhost/algo.jpg'
  };

  beforeEach(() => {
    wrapper = shallow(<GifGridItem {...props} />);
  });

  test('should be render <GifGridItem />', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should be render a paragraph with title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(props.title);
  });

  test('should be have image equal to props', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(props.url);
    expect(img.prop('alt')).toBe(props.title);
  });

  test('should be have card class', () => {
    const div = wrapper.find('div');
    expect(div.hasClass('card')).toBe(true);
  });
});
