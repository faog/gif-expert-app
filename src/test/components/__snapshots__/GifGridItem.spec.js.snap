// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem /> Component should be render <GifGridItem /> 1`] = `
<div
  className="card"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
